tests/keras/layers/convolutional_test.py

```diff
@@ -541,6 +529,26 @@ def test_convolution_3d(padding, strides):
                             stack_size))
 
 
+@keras_test
+@pytest.mark.skipif((K.backend() == 'cntk'),
```

Comment:
The tests won't run on CNTK, even if there is a GPU. I know it was also the cases with other tests before this PR, but it would make the life of maintainers easier since we could quickly run the file and get the results.

Do you think that would be possible to check if the tests are run on the gpu or not and skip if necessary?
I see some kind of check in the cntk backend like this. Since I don't have a lot of knowledge of cntk, I'm guessing: `if dev.type() == 0: ...`

Reply:
Thanks for your review. If we want to check whether the tests run on GPU or not, we should implement function for each backend, and wrap them together as a single interface. You can refer [```_get_available_devices```](https://github.com/keras-team/keras/blob/master/keras/utils/multi_gpu_utils.py#L15), that is only for TF backend. If this is really necessary, I can send a separate follow-up PR to add it, as several other places have this issue, we can sweep them together.
Actually the reason that we don't find this hidden bug before is we only run tests and check output expected for number of channel is 1, we don't have unit tests for number of channel is great than 1. And this bug only happened when number of channels is great than 1.

Reply:
I agree that it would be nice to have a cross backend way to do this. For the moment, I propose to do (since this tests issue is only limited to cntk):
```python
@pytest.mark.skipif((K.backend() == 'cntk' and K.dev.type() == 0),
```

Feel free to add any tests you see fit. It'd be nice if we could prevent this bug from happening again.